### DISCUSSION TITLE
Query doesn't re-run when variable changes

### DISCUSSION THREAD

**Discussion by bmccorm2** [urql-graphql/urql#2247] [Q&A]
Hi - I have a query where i supply an ID but when that ID changes the query does not re-run.  I have a Svelte app and am using svelte-spa-router to get the variable from the URL.  I have an anchor tag elsewhere in the page where the user can navigate to a different CarId.  I can `console.log(params.carId)` and see that it changes, however the query is never re-run.  Am i doing something wrong?

```
  export let params = {};
  const res = operationStore(GET_CONSUMPTION, {
    carId: parseInt(params.carId),
  });
  query(res);
```

**Comment by thojanssens** [comment]
Hello, I have the same question, can we get in touch and help each other out? Are you in the Svelte Discord server maybe?

**Comment by bmccorm2** [comment]
@thojanssens - sorry i didn't follow up on this.  This is what i ended up using.  Written this way, Svelte will react to changes in carId which will prompt a new query run.  Let me know if that doesn't help you and i can DM on discord.

```
  const res = operationStore(GET_CONSUMPTION, { carId });
  $: res.variables = { carId };
  query(res);
```
